src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/BalanceCalculatorConsumerServiceTest.java

```diff
@@ -0,0 +1,144 @@
+        id, ownerId);
+
+    //when
+    balanceCalculatorConsumerService.onMessage(message);
+
+    // then
+    assertThat(capturedOutput.getOut()).contains(String.format(KAFKA_SUCCESSFUL_DESERIALIZATION,
+        expectedTimeEntry.getId()));
+  }
+
+  //Invalid resource throws error
+  @Test
+  void onMessage_notDeserializeKafkaMessageAndThrowException_when_inValidResourceIsReceived() throws JsonProcessingException {
+    //given
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_INVALID_RESOURCE, MESSAGE_VALID_VERSION
+        , id, ownerId);
+
+
+    assertThatThrownBy(() -> {
+     balanceCalculatorConsumerService.onMessage(message);
+    }).isInstanceOf(KafkaConsumerException.class)
+        .hasMessageContaining(String.format(KAFKA_UNSUCCESSFUL_DESERIALIZATION, message));
+
+  }
+
+  //Invalid version throws error
+  @Test
+  void onMessage_notDeserializeKafkaMessageAndThrowException_inValidVersionIsReceived() throws JsonProcessingException {
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_INVALID_VERSION
+        , id, ownerId);
+
+    assertThatThrownBy(() -> {
+      balanceCalculatorConsumerService.onMessage(message);
+    }).isInstanceOf(KafkaConsumerException.class)
+        .hasMessageContaining(String.format(KAFKA_SCHEMA_INVALID_VERSION, message));
+  }
+
+  //Desearilization error (missing/extra field?) on resource
+  @Test
+  void onMessage_notDeserializeKafkaMessage_extraFieldReceived() throws JsonProcessingException {
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode resource = mapper.createObjectNode();
+    resource.put("id", id);
+    resource.put("tenantId", VALID_TENANT_ID);
+    resource.put("ownerId", ownerId);
+    resource.put("timePeriodTypeId", VALID_TIME_PERIOD_TYPE_ID);
+    resource.put("shiftType", " ");
+    resource.put("actualStartTime", VALID_START_TIME);
+    resource.put("actualEndTime", VALID_END_TIME);
+    resource.put("extraField", "EXTRA_FIELD");
+
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_VALID_VERSION,
+        resource);
+
+    assertThatThrownBy(() -> {
+      balanceCalculatorConsumerService.onMessage(message);
+    }).isInstanceOf(KafkaConsumerException.class)
+        .hasMessageContaining(String.format(KAFKA_UNSUCCESSFUL_DESERIALIZATION, message));
+  }
+
+  //Invalid date format received
+  @Test
+  void onMessage_notDeserializeKafkaMessage_invalidDateFormatReceived() throws JsonProcessingException {
+    String id = UUID.randomUUID().toString();
+    String ownerId = UUID.randomUUID().toString();
+
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
make this mapper a class member so it doesn't have to be instantiated for different tests